From 4f3bce4bf75ec43c82fe662db671973293336145 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Tue, 26 Oct 2021 17:19:20 +0300
Subject: [PATCH] [SW-56687] gaudi2: pldm: bump up cpu pll

In order to get uart prints on pldm, bump up cpu pll frequency.

Change-Id: I085b0bb8de46f9ea943640d1f68161d10a72025d
---

diff --git a/arch/gaudi2/common/pldm.c b/arch/gaudi2/common/pldm.c
index d45b703..9f930b1 100644
--- a/arch/gaudi2/common/pldm.c
+++ b/arch/gaudi2/common/pldm.c
@@ -5,8 +5,11 @@
  *
  */
 
+#include <gaudi2_blocks_linux_driver.h>
 #include <preboot.h>
 #include <psoc_global_conf_regs.h>
+#include <psoc_cpu_pll_ctrl_regs.h>
+#include <pll.h>
 #include <uart.h>
 
 
@@ -19,8 +22,6 @@
  * This cfg specific to PLDM bumps up uart freq
  * to 250 MHz while keeping other div's on refclk
  */
-/*
- * TODO: Enable this later
 static struct hl_pll_cfg gaudi2_pll_cfg[] = {
 	{
 		.name = "CPU", .base = mmPSOC_CPU_PLL_CTRL_BASE,
@@ -33,7 +34,6 @@
 		},
 	}
 };
-*/
 
 u64 hl_pll_timestamp_freq_get(void)
 {
@@ -63,9 +63,8 @@
 
 void hl_dev_pll_init(void)
 {
-	/* TODO: Enable this later */
 	/* Bump up UART frequency */
-	//hl_pll_init(gaudi2_pll_cfg, 1);
+	hl_pll_init(gaudi2_pll_cfg, 1);
 }
 
 void preboot_init_pcie_phy(void) {}
